src/api/messages/getTemporaryFileUrl.js

```diff
@@ -0,0 +1,29 @@
+/* @flow strict-local */
+import type { Auth, ApiResponseSuccess } from '../transportTypes';
+import { apiGet } from '../apiFetch';
+
+type ApiResponseTempFileUrl = {|
+  ...ApiResponseSuccess,
+  url: string,
+|};
+
+/**
+ * Get a temporary, authless partial URL to a realm-uploaded file.
+ *
+ * This endpoint was introduced in zulip/zulip@9e72fbe3f. The URL returned
+ * allows a file to be viewed immediately without requiring a fresh login, but
+ * it also does not include secrets like the API key. Currently, this URL
+ * remains valid for 60 seconds. It can be used in various places in the app
+ * where we open files in the browser, saving user's time.
+ *
+ * @param auth Authentication info of the current user.
+ * @param filePath The partial location of the uploaded file, as provided
+ * by the Zulip server to us in the message containing the file.
+ * It is a string of the form `/user_uploads/{realm_id_str}/{filename}`.
+ * This enpoint is not documented, but the zulip.yaml[1] file describes
+ * this endpoint in brief.
+ *
+ * [1] https://github.com/zulip/zulip/blob/master/zerver/openapi/zulip.yaml.
```

Comment:
nit: indent overflow of `@param`

Oh actually and more than a nit! Looking closer I see that what looks like one paragraph describing `filePath` is really two, and the second one isn't related to the specific parameter at all.

That should have a paragraph break. :-) And the stuff that isn't about a particular parameter should go before the `@param` items -- otherwise it gets parsed, both by the machine (when you hover on a call site) and potentially by the human reader, as part of the parameter description.

Reply:
Fixed.